src/ott/neural/flows/genot.py

```diff
@@ -0,0 +1,490 @@
+    k_samples_per_x: Number of samples drawn from the conditional distribution
+      of an input sample, see algorithm TODO.
+    solver_latent_to_data: Linear OT solver to match the latent distribution
+      with the conditional distribution.
+    kwargs_solver_latent_to_data: Keyword arguments for `solver_latent_to_data`.
+      #TODO: adapt
+    fused_penalty: Fused penalty of the linear/fused term in the Fused
+      Gromov-Wasserstein problem.
+    tau_a: If :math:`<1`, defines how much unbalanced the problem is
+    on the first marginal.
+    tau_b: If :math:`< 1`, defines how much unbalanced the problem is
+    on the second marginal.
+    rescaling_a: Neural network to learn the left rescaling function. If
+      :obj:`None`, the left rescaling factor is not learnt.
+    rescaling_b: Neural network to learn the right rescaling function. If
+      :obj:`None`, the right rescaling factor is not learnt.
+    unbalanced_kwargs: Keyword arguments for the unbalancedness solver.
+   callback_fn: Callback function.
+    rng: Random number generator.
+  """
+
+  def __init__(
+      self,
+      velocity_field: Callable[[
+          jnp.ndarray, jnp.ndarray, Optional[jnp.ndarray], Optional[jnp.ndarray]
+      ], jnp.ndarray],
+      input_dim: int,
+      output_dim: int,
+      cond_dim: int,
+      iterations: int,
+      valid_freq: int,
+      ot_solver: Type[was_solver.WassersteinSolver],
+      epsilon: float,
+      cost_fn: Union[costs.CostFn, Dict[str, costs.CostFn]],
+      scale_cost: Union[Union[bool, int, float,
+                              Literal["mean", "max_norm", "max_bound",
+                                      "max_cost", "median"]],
+                        Dict[str, Union[bool, int, float,
+                                        Literal["mean", "max_norm", "max_bound",
+                                                "max_cost", "median"]]]],
+      optimizer: Type[optax.GradientTransformation],
+      flow: Type[BaseFlow] = ConstantNoiseFlow(0.0),
+      time_sampler: Callable[[jax.Array, int], jnp.ndarray] = sample_uniformly,
+      checkpoint_manager: Type[checkpoint.CheckpointManager] = None,
+      k_samples_per_x: int = 1,
+      solver_latent_to_data: Optional[Type[was_solver.WassersteinSolver]
+                                     ] = None,
+      kwargs_solver_latent_to_data: Dict[str, Any] = types.MappingProxyType({}),
+      fused_penalty: float = 0.0,
+      tau_a: float = 1.0,
+      tau_b: float = 1.0,
+      rescaling_a: Callable[[jnp.ndarray], float] = None,
+      rescaling_b: Callable[[jnp.ndarray], float] = None,
+      unbalanced_kwargs: Dict[str, Any] = types.MappingProxyType({}),
+      callback_fn: Optional[Callable[[jnp.ndarray, jnp.ndarray, jnp.ndarray],
+                                     Any]] = None,
+      rng: Optional[jax.Array] = None,
+  ):
+    rng = utils.default_prng_key(rng)
+    rng, rng_unbalanced = jax.random.split(rng)
+    BaseNeuralSolver.__init__(
```

Comment:
This is a wrong usage of the init for mixins, please refactor using composition over inheritance.